gui/src/components/health/cortx-health-graphical.vue

```diff
@@ -0,0 +1,544 @@
+  public clusterHealthData: any = {};
+
+  public mounted() {
+    this.getHealthData();
+  }
+
+  public async getHealthData() {
+    this.healthQueryParams = {
+      depth: 0
+    };
+    this.$store.dispatch("systemConfig/showLoader", "Fetching health...");
+    const res = await Api.getAll(apiRegister.health_cluster,
+      this.healthQueryParams
+    );
+    this.clusterHealthData = res.data;
+
+    function traverse(obj) {
+      if (obj.sub_resources && obj.sub_resources.length > 0) {
+        rename(obj);
+        if (obj.children.length > 0) {
+          for (let i = 0; i < obj.children.length; i++) {
+            traverse(obj.children[i]);
+          }
+        }
+      } else {
+        rename(obj);
+      }
+    }
+
+    function rename(obj) {
+      obj.children = obj.sub_resources;
+      delete obj.sub_resources;
+    }
+
+    traverse(this.clusterHealthData.data[0]);
+
+    this.$store.dispatch("systemConfig/hideLoader");
+
+    this.plotTree();
+  }
+
+  public plotTree() {
+    // Set the dimensions and margins of the diagram
+    const margin = { top: 20, right: 90, bottom: 30, left: 90 };
+    const width = 1000 - margin.left - margin.right;
+    const height = 500 - margin.top - margin.bottom;
+    // append the svg object to the body of the page
+    // appends a 'group' element to 'svg'
+    // moves the 'group' element to the top left margin
+    const svg = d3
+      .select("#treeContainer")
+      .append("svg")
+      .attr("width", width + margin.right + margin.left)
+      .attr("height", height + margin.top + margin.bottom)
+      .style("overflow", "auto")
+      .append("g")
+      .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+    const i = 0,
+      duration = 750;
+    // declares a tree layout and assigns the size
+    const treemap = d3.tree().size([height, width]);
+    // Assigns parent, children, height, depth
+    const root = d3.hierarchy(this.clusterHealthData.data[0], function(d: any) {
+      return d.children;
+    });
+    root.x0 = 0;
+    root.y0 = 0;
+    // Collapse after the second level
+    root.children.forEach(collapse);
+    update(root);
+    // Collapse the node and all it's children
+    function collapse(d: any) {
+      if (d.children && d.depth > 2) {
+        d._children = d.children;
+        d._children.forEach(collapse);
+        d.children = null;
+      } else if (d.children) {
+        d.children.forEach(collapse);
+      }
+    }
+    function update(source) {
+      // Assigns the x and y position for the nodes
+      let treeData = treemap(root);
+      // Compute the new tree layout.
+      let nodes = treeData.descendants(),
+        links = treeData.descendants().slice(1);
+      // Normalize for fixed-depth.
+      nodes.forEach(function(d: any) {
+        d.y = d.depth * 180;
+      });
+      // ****************** Nodes section ***************************
+      // Update the nodes...
+      let i = 0;
```

Comment:
Is this variable used anywhere? If not please remove it.

Reply:
Yes it's required.